stdlib/public/Python/NumpyConversion.swift

```diff
@@ -28,128 +28,74 @@ public protocol ConvertibleFromNumpyArray {
 
 /// A type that is compatible with a NumPy scalar `dtype`.
 public protocol NumpyScalarCompatible {
+  /// Whether this type is compatible with the given Numpy scalar `dtype`.
   static func isCompatible(withNumpyScalarType dtype: PythonObject) -> Bool
+
+  /// A single NumPy scalar `dtype` that this type is compatible with.
+  static var numpyScalarType: PythonObject { get }
```

Comment:
I see the problem with the inconsistency that this allows. I do not think a second protocol solves it -- often, the protocols will be used together, and inconsistencies between them will cause problems. e.g. in the csv dataset initializer, it uses both and it will fail if the dtype passes to `np.loadtxt` doesn't satisfy the `isCompatible(withNumpyScalarType:)` that `Tensor(numpyArray:)` makes.

I can make inconsistency impossible by changing the protocol to have a single static array requirement. So I did that. Do you think that is a good solution?

Reply:
We chatted in person and concluded that making the single protocol require a computed property is a great idea!